### PR TITLE
Add ability to configure custom_trusted_oid_mapping.yaml

### LINKED ISSUE #602: /etc/puppetlabs/puppet/custom_trusted_oid_mapping.yaml should be manageable
In setups which use trusted facts, /etc/puppetlabs/puppet/custom_trusted_oid_mapping.yaml needs to be configured to map the OID to a usable variable name.

See https://puppet.com/docs/puppet/5.3/ssl_attributes_extensions.html#extension-requests-permanent-certificate-data for more information

### PR DESCRIPTION
**Pull request by baurmatt**
This commit adds support to configure the
$confdir/custom_trusted_oid_mapping.yaml as described in
https://puppet.com/docs/puppet/5.5/config_file_oid_map.html.

Fixes GH-602

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -182,7 +182,7 @@
 # $environment::                            Default environment of the Puppet agent
 #
 # $agent_additional_settings::              A hash of additional agent settings.
-#                                           Example: {stringify_facts => true}
+#                                           Example: {stringify_facts                          => true}
```
Comment (ekohl) [inline manifests/init.pp]: Could you revert this change and other whitespace only changes?

File: types/custom_trusted_oid_mapping.pp [hunk 2]
```diff
@@ -0,0 +1,4 @@
+type Puppet::Custom_trusted_oid_mapping = Hash[String, Struct[{
```
Comment (ekohl) [inline types/custom_trusted_oid_mapping.pp]: We have a bug in our installer that it currently doesn't handle multi line types because it does incorrect manual parsing. We should be fixing it but in the short term I wonder if we should put it on a single line.

File: spec/classes/puppet_server_config_spec.rb [hunk 3]
```diff
@@ -832,6 +836,29 @@
...
+                },
+                '1.3.6.1.4.1.34380.1.2.1.2' => {
+                  shortname => 'myothershortname',
+                },
+              }
+           }"
+        end
+
+        it 'should have a configured custom_trusted_oid_mapping.yaml' do
+          should contain_file("#{confdir}/custom_trusted_oid_mapping.yaml").
```
Comment (ekohl) [inline spec/classes/puppet_server_config_spec.rb]: We have a `verify_exact_contents` helper which might be easier to use.

File: manifests/init.pp (hunk 1 above)
Comment (baurmatt) [reply manifests/init.pp]: Done! Sorry! :/ Are the whitespace changes in server.pp ok?

File: types/custom_trusted_oid_mapping.pp (hunk 2 above)
Comment (baurmatt) [reply types/custom_trusted_oid_mapping.pp]: Sure!

File: manifests/init.pp (hunk 1 above)
Comment (ekohl) [reply manifests/init.pp]: Yes, indenting is no problem.

File: spec/classes/puppet_server_config_spec.rb (hunk 3 above)
Comment (baurmatt) [reply spec/classes/puppet_server_config_spec.rb]: This is really helpful, thanks! :)

**Comment by baurmatt** [comment]
Fixed review comments, thanks! :)

**Comment by ekohl** [comment]
Thanks!
